src/lib.rs

```diff
@@ -427,19 +426,16 @@ pub struct PrometheusMetricsMiddleware<S> {
     inner: Arc<PrometheusMetrics>,
 }
 
-impl<S, B> Service for PrometheusMetricsMiddleware<S>
+impl<S, B> Service<ServiceRequest> for PrometheusMetricsMiddleware<S>
 where
-    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
+    S: Service<ServiceRequest, Response = ServiceResponse<B>, Error = Error>,
     B: MessageBody,
 {
-    type Request = ServiceRequest;
     type Response = ServiceResponse<StreamLog<B>>;
     type Error = S::Error;
     type Future = LoggerResponse<S, B>;
 
-    fn poll_ready(&mut self, ct: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        self.service.poll_ready(ct)
-    }
+    actix_service::always_ready!();
```

Comment:
this macro expands to `Poll::Ready(Ok(()))` which is not we want to have here, I think)

Reply:
It was without the always_ready but was not working anyway in runtime, so i changed to it because i see it in the changelog of actix_service

Reply:
But still not working, i talked about it in the gitter chat of actix and they said that is needed more changes that only the service trait modifications but i did not found what

Reply:
I run into the same issue - need actix_wev_prom for tokio1.0. So create pull request too. But mine is compile and tests passed. 
So you can use it ;)
https://github.com/nlopes/actix-web-prom/pull/45

Reply:
By branch was like yours and still not worked thats why i made the change that i just roll_back, i think i have some other thing that is causing this error since it is working for you

Reply:
Are you calling it by web::Data<IntCounterVec> or did it changed?

Reply:
Found it, i was using two versions of prometheus of my workspace.